### PR TITLE
Add @tsconfig/node20, add node 20 to test matrix

### PR DESCRIPTION
**Pull request by chentsulin**
Add @tsconfig/node20, add node 20 to test matrix.

### CODE REVIEW

**Comment by RobinTail** [comment]
Does this fix https://github.com/TypeStrong/ts-node/issues/1997 (the ESM support) , @chentsulin ?

**Comment by R-Rajaneesh** [comment]
+1 Bump

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -87,10 +87,21 @@ jobs:
             nodeFlag: 18
             typescript: next
             typescriptFlag: next
-          # Node nightly
+          # Node 20
```
Comment (cspotcode) [inline .github/workflows/continuous-integration.yml:90]: also need to add to `flavor` array above

File: .github/workflows/continuous-integration.yml [hunk 2]
```diff
@@ -87,10 +87,21 @@ jobs:
...
-            nodeFlag: 20_nightly
+            node: 20
+            nodeFlag: 20
+            typescript: latest
+            typescriptFlag: latest
+          - flavor: 6
+            node: 20
+            nodeFlag: 20
+            typescript: next
+            typescriptFlag: next
```
Comment (cspotcode) [inline .github/workflows/continuous-integration.yml:100]: We can remove this one; I'm trying to keep only a single entry for typescript@next.  I suspect that github actions throttle if you use it too much, so I try to keep the test matrix small where possible.

**Comment by cspotcode** [comment]
I renamed this PR to clarify that it does not add support for node 20, it only adds the @tsconfig/bases for node 20.

**Comment by R-Rajaneesh** [comment]
@cspotcode So when will ts-node support node v20, my projects are currently in a standstill due to this

File: .github/workflows/continuous-integration.yml [hunk 3]
```diff
@@ -87,10 +87,16 @@ jobs:
...
-            node: 20-nightly
-            nodeFlag: 20_nightly
+            node: 20
+            nodeFlag: 20
+            typescript: latest
+            typescriptFlag: latest
+          # Node nightly
+          - flavor: 6
+            node: nightly
+            nodeFlag: nightly
```
Comment (cspotcode) [inline .github/workflows/continuous-integration.yml:99]:
```suggestion
            node: 21-nightly
            nodeFlag: 21_nightly
```

**Comment by OneCricketeer** [comment]
Should there be a release newer than a year ago that makes mention on Node20? 

https://github.com/TypeStrong/ts-node/releases

**Pull request merged**